Simple FileEditor crashes with path

When I run the example: https://github.com/enthought/traitsui/blob/master/examples/demo/Standard_Editors/FileEditor_demo.py, using the simple no-history editor (the top one), I get a crash with the following sequence:

1. Click the folder icon
2. Select a file
3. Click "Open" (the full path appears in the text box)
4. Click the folder icon
5. Crash:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/traitsui/wx/file_editor.py", line 183, in show_file_dialog
    dlg       = self._create_file_dialog()
  File "/usr/lib/python2.7/dist-packages/traitsui/wx/file_editor.py", line 263, in _create_file_dialog
    dlg.SetFilename( self._get_value() )
  File "/usr/lib/python2.7/dist-packages/wx-3.0-gtk2/wx/_windows.py", line 3173, in SetFilename
    return _windows_.FileDialog_SetFilename(*args, **kwargs)
wx._core.PyAssertionError: C++ assertion "volDummy.empty() && pathDummy.empty()" failed at ../src/common/filename.cpp(568) in Assign(): the file name shouldn't contain the path
```

It doesn't crash if the text box contains only the filename, without the path, but this won't be possible if I want to select a file not in the current directory.

This is probably fixed in newer versions of python/wx/traits (I'm using Ubuntu 16.04), but maybe there's a workaround I can use?

--- a/traitsui/wx/file_editor.py
+++ b/traitsui/wx/file_editor.py
@@ -25,7 +25,7 @@
 import wx
 
 from os.path \
-    import abspath, splitext, isfile, exists
+    import abspath, split, splitext, isfile, exists
 
 from traits.api \
     import List, Str, Event, Any, on_trait_change, TraitError
@@ -258,12 +258,16 @@ class SimpleEditor(SimpleTextEditor):
         else:
             style = wx.FD_DEFAULT_STYLE
 
-        dlg = wx.FileDialog(self.control,
-                            message='Select a File',
-                            wildcard=wildcard,
-                            style=style)
+        directory, filename = split(self._get_value())
 
-        dlg.SetFilename(self._get_value())
+        dlg = wx.FileDialog(
+            self.control,
+            defaultDir=directory,
+            defaultFile=filename,
+            message='Select a File',
+            wildcard=wildcard,
+            style=style
+        )
 
         return dlg
 

